src/pages/RegisterPage.js

```diff
@@ -0,0 +1,201 @@
+import React, { useRef } from "react";
+import { useForm } from "react-hook-form";
+import "../assets/styles/RegisterForm.css";
+import { PrimaryButton } from "../components/PrimaryButton";
+import axios from "axios";
+
+function RegisterPage() {
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    watch,
+  } = useForm({ mode: "onBlur" });
+
+  const password = useRef({});
+  password.current = watch("password", "");
+
+  const onSubmit = (data, e) => {
+    e.target.reset();
+    console.log(data);
+    axios
+      .post("https://jsonplaceholder.typicode.com/posts", data)
+      .then(response => console.log(response))
+      .catch(error => console.log(error));
+  };
+
+  return (
+    <section className="register">
+      <div className="register__container">
+        <h2>Sign up to continue:</h2>
+        <form
+          onSubmit={handleSubmit(onSubmit)}
+          className="register__container--form"
+        >
+          {errors?.firstName?.type === "required" && (
+            <p>*First name is required</p>
+          )}
+          {errors?.firstName?.type === "maxLength" && (
+            <p>*First name cannot exceed 20 characters</p>
+          )}
+          {errors?.firstName?.type === "pattern" && (
+            <p>*Alphabetical characters only</p>
+          )}
+          <input
+            type="text"
+            name="firstName"
+            placeholder="First name"
+            className="form__field"
+            {...register("firstName", {
+              required: true,
+              maxLength: 20,
+              pattern: /^[A-Za-z]+$/i,
+            })}
+          />
+
+          {errors?.lastName?.type === "required" && (
+            <p>*Last name is required</p>
+          )}
+          {errors?.lastName?.type === "maxLength" && (
+            <p>*First name cannot exceed 20 characters</p>
+          )}
+          {errors?.lastname?.type === "pattern" && (
+            <p>*Alphabetical characters only</p>
+          )}
+          <input
+            type="text"
+            placeholder="Last name"
+            name="lastName"
+            className="form__field"
+            {...register("lastName", {
+              required: true,
+              maxLength: 20,
+              pattern: /^[A-Za-z]+$/i,
+            })}
+          />
+
+          {errors?.email?.type === "required" && <p>*Email is required</p>}
+          {errors?.email?.type === "pattern" && (
+            <p>*Please enter a valid email.</p>
+          )}
+          <input
+            type="email"
+            placeholder="Email"
+            name="email"
+            className="form__field"
+            {...register("email", {
+              required: true,
+              pattern: /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/,
+            })}
+          />
+
+          {/* { {errors?.address?.type === 'required' && <p>*Address is required</p>}
```

Comment:
Please remove this commented code

Reply:
Thank you Leonel! I removed unnecessary commented code.